### PR TITLE
[CI] Add downstream builds

### PR DESCRIPTION
**Pull request by christophfroehlich**
- run them only in PRs
- build+test downstream ros-controls packages
- build-only some downstream 3rd party packages. any more we should add?

### CODE REVIEW

File: .github/workflows/rolling-semi-binary-downstream-build.yml
Comment (saikishor) [inline .github/workflows/rolling-semi-binary-downstream-build.yml:54]: Why is it here formatted differently compared to Humble and Jazzy?

Comment (christophfroehlich) [reply .github/workflows/rolling-semi-binary-downstream-build.yml]: you mean due to the matrix strategy? this makes it easier to add a new distro before we branch for it. I created already distinct files for jazzy, but kept this for the future

**Comment by christophfroehlich** [comment]
I submitted a PR to fix the failing tests to https://github.com/PickNikRobotics/topic_based_ros2_control/pull/30

**Comment by saikishor** [comment]
> I submitted a PR to fix the failing tests to [PickNikRobotics/topic_based_ros2_control#30](https://github.com/PickNikRobotics/topic_based_ros2_control/pull/30)

@christophfroehlich Should we wait until this is merged?

**Comment by christophfroehlich** [comment]
> > I submitted a PR to fix the failing tests to [PickNikRobotics/topic_based_ros2_control#30](https://github.com/PickNikRobotics/topic_based_ros2_control/pull/30)
> 
> @christophfroehlich Should we wait until this is merged?

fixed now

**Review by fmauch — APPROVED** [review]
Definitively a plus that we know early when something breaks :-)

**Pull request merged**
